mysql/resource_ti_config_variable_test.go

```diff
@@ -0,0 +1,213 @@
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+			{
+				Config:      testAccConfigVarConfig_withInstanceAndType("varName", varValue, varType, getGetInstance(varType, t)),
+				ExpectError: regexp.MustCompile(".*Error: bad request to:*"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccConfigVarExists(varName, varValue, varType),
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+			{
+				Config:      testAccConfigVarConfig_withInstanceAndType(varName, varValue, "varType", getGetInstance(varType, t)),
+				ExpectError: regexp.MustCompile(".*Error: expected type to be one of.*"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccConfigVarExists(varName, varValue, varType),
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+		},
+	})
+}
+
+func TestTiKvConfigVar_basic(t *testing.T) {
+	varName := "split.qps-threshold"
+	varValue := "1000"
+	varType := "tikv"
+	varInstance := ""
+	resourceName := "mysql_ti_config.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccPreCheckSkipNotTiDB(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccConfigVarCheckDestroy(varName, varType),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccConfigVarConfig_basic(varName, varValue, varType),
+				Check: resource.ComposeTestCheckFunc(
+					testAccConfigVarExists(varName, varValue, varType),
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+			{
+				Config: testAccConfigVarConfig_withInstanceAndType(varName, varValue, varType, varInstance),
+				Check: resource.ComposeTestCheckFunc(
+					testAccConfigVarExists(varName, varValue, varType),
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+			{
+				Config:      testAccConfigVarConfig_withInstanceAndType(varName, "varValue", varType, varInstance),
+				ExpectError: regexp.MustCompile(".*Error: error setting value*"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccConfigVarExists(varName, "varValue", varType),
+					resource.TestCheckResourceAttr(resourceName, "name", varName),
+				),
+			},
+		},
+	})
+}
+
+func testAccConfigVarExists(varName string, varValue string, varType string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		db, err := connectToMySQL(testAccProvider.Meta().(*MySQLConfiguration))
+		if err != nil {
+			return err
+		}
+
+		resName, resValue, err := testAccGetConfigVar(varName, varType, db)
+
+		if err != nil {
+			return err
+		}
+
+		if resValue == varValue {
+			return nil
+		}
+
+		return fmt.Errorf("variable '%s' not found. resName: %s, resValue: %s, err: %s", varName, resName, resValue, err)
+	}
+}
+
+func getGetInstance(varType string, t *testing.T) string {
+	var resInstanceType, resInstance, resName, resValue string
+	match, _ := regexp.MatchString("^(mysql:).*", os.Getenv("DB"))
```

Comment:
I don't really understand - how is `DB` environment set?

Reply:
With Travis tests Here https://github.com/petoju/terraform-provider-mysql/blob/master/.travis.yml#L11-L13

Reply:
Ah, ok then - but could you add a comment? Because that's completely opaque you want to skip travis runs.